src/chain/chain.rs

```diff
@@ -662,6 +662,12 @@ impl Chain {
             .map(|member_info| member_info.p2p_node.connection_info())
     }
 
+    /// Returns the `P2pNode` for a member of our section.
+    pub fn get_member_p2p_node_by_id(&self, pub_id: &PublicId) -> Option<&P2pNode> {
+        self.get_member_p2p_node(pub_id.name())
+            .filter(|p2p_node| p2p_node.public_id() == pub_id)
```

Comment:
Is the filter necessary?
And if not, not sure it really help compared to `self.get_member_p2p_node(pub_id.name())`

Reply:
Yeah good question, is it possible to happen? That the duplication is inconsistent?